pkg/skaffold/kubernetes/portforward/kubectl_forwarder.go

```diff
@@ -70,11 +70,13 @@ var (
 // It kills the command on errors in the kubectl port-forward log
 // It restarts the command if it was not cancelled by skaffold
 // It retries in case the port is taken
-func (k *KubectlForwarder) Forward(parentCtx context.Context, pfe *portForwardEntry) {
-	go k.forward(parentCtx, pfe)
+func (k *KubectlForwarder) Forward(parentCtx context.Context, pfe *portForwardEntry) error {
+	errChan := make(chan error, 1)
+	go k.forward(parentCtx, pfe, errChan)
+	return <-errChan
```

Comment:
Doesn't `return <-errChan` necessarily make `Forward` blocking?

Reply:
Yes, now Forward will block and not return until a message has been received. The only thing that could potentially be blocked in execution though would be the port forwarding of resources as those all run together in their own goroutine. Otherwise execution would proceed as normal, and if anything goes wrong with port forwarding that will be conveyed to the user.

Reply:
When you say "Execution will proceed as normal" ,does that mean, logs will be streamed?
We want to make sure, port-forward does not block streaming logs. 

On master, you can see logs been streamed even tough a port-forward failed.
```
DEBU[0003] Found dependencies for dockerfile: []        
Watching for changes...
INFO[0003] Streaming logs from pod: leeroy-web-5df95c76c5-nnn5t container: leeroy-web 
DEBU[0003] Running command: [kubectl --context minikube logs --since=4s -f leeroy-web-5df95c76c5-nnn5t -c leeroy-web --namespace default] 
INFO[0003] Streaming logs from pod: leeroy-app-595989447b-98rcq container: leeroy-app 
DEBU[0003] Running command: [kubectl --context minikube logs --since=4s -f leeroy-app-595989447b-98rcq -c leeroy-app --namespace default] 
DEBU[0003] Forwarding service leeroy-app/50051 to pod leeroy-app-595989447b-98rcq/50051 
DEBU[0003] Running command: [kubectl --context minikube port-forward --pod-running-timeout 1s --namespace default pod/leeroy-app-595989447b-98rcq 50053:50051] 
DEBU[0003] port forwarding deployment-leeroy-wb-default-8080 got terminated: exit status 1, output: Error from server (NotFound): deployments.apps "leeroy-wb" not found 
[leeroy-app] 2020/11/02 17:05:20 leeroy app server ready
[leeroy-web] 2020/11/02 17:05:20 leeroy web server ready
DEBU[0004] Running command: [kubectl --context minikube port-forward --pod-running-timeout 1s --namespace default deployment/leeroy-wb 9000:8080] 
DEBU[0004] port forwarding deployment-leeroy-wb-default-8080 got terminated: exit status 1, output: Error from server (NotFound): deployments.apps "leeroy-wb" not found 
DEBU[0004] Executing template &{envTemplate 0xc00046e400 0xc000ea7180  } with environment map[HOME:/Users/tejaldesai LANG:en_US.UTF-8 LESS:-R LOGNAME:tejaldesai LSCOLORS:Gxfxcxdxbxegedabagacad OLDPWD:/Users/tejaldesai/workspace/skaffold PAGER:less PATH:/Users/tejaldesai/Downloads/google-cloud-sdk2/bin:/Users/tejaldesai/mdproxy/bin:/Users/tejaldesai/Downloads/google-cloud-sdk2/bin:/Users/tejaldesai/mdproxy/bin:/usr/local/git/current/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/local/go/bin PWD:/Users/tejaldesai/workspace/skaffold/examples/microservices SHELL:/bin/zsh SHLVL:1 SK_SIGNING_PLUGIN:gnubbyagent SSH_AUTH_SOCK:/private/tmp/com.apple.launchd.LzK5Ha5WzO/Listeners TERM:xterm-256color TERM_PROGRAM:Apple_Terminal TERM_PROGRAM_VERSION:433 TERM_SESSION_ID:0405193E-BCF1-4630-96CD-9173EF808119 TMPDIR:/var/folders/gk/s778hvkj0lb0zl95ywvw0snr00cfc0/T/ USER:tejaldesai XPC_FLAGS:0x0 XPC_SERVICE_NAME:0 ZSH:/Users/tejaldesai/.oh-my-zsh _:/Users/tejaldesai/workspace/skaffold/examples/microservices/../../out/skaffold] 
DEBU[0004] Executing template &{envTemplate 0xc00046e500 0xc000ea7680  } with environment map[HOME:/Users/tejaldesai LANG:en_US.UTF-8 LESS:-R LOGNAME:tejaldesai LSCOLORS:Gxfxcxdxbxegedabagacad OLDPWD:/Users/tejaldesai/workspace/skaffold PAGER:less PATH:/Users/tejaldesai/Downloads/google-cloud-sdk2/bin:/Users/tejaldesai/mdproxy/bin:/Users/tejaldesai/Downloads/google-cloud-sdk2/bin:/Users/tejaldesai/mdproxy/bin:/usr/local/git/current/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/local/go/bin PWD:/Users/tejaldesai/workspace/skaffold/examples/microservices SHELL:/bin/zsh SHLVL:1 SK_SIGNING_PLUGIN:gnubbyagent SSH_AUTH_SOCK:/private/tmp/com.apple.launchd.LzK5Ha5WzO/Listeners TERM:xterm-256color TERM_PROGRAM:Apple_Terminal TERM_PROGRAM_VERSION:433 TERM_SESSION_ID:0405193E-BCF1-4630-96CD-9173EF808119 TMPDIR:/var/folders/gk/s778hvkj0lb0zl95ywvw0snr00cfc0/T/ USER:tejaldesai XPC_FLAGS:0x0 XPC_SERVICE_NAME:0 ZSH:/Users/tejaldesai/.oh-my-zsh _:/Users/tejaldesai/workspace/skaffold/examples/microservices/../../out/skaffold] 
DEBU[0004] Found dependencies for dockerfile: [{web.go /go true}] 



```

Reply:
Yeah logs will continue streaming whether it fails or succeeds, or is blocked.

Reply:
perfect!